File: KthNumber/src/main/kotlin/KthNumber.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = {};

        int[] a = new int[commands.length];
        int a_index = 0;
        for(int p = 0; p < commands.length; p++)
        {
            int length1 = commands[p][1] - commands[p][0] +1 ;
            int[] temp = new int[length1];
            int cnt = 0;
            for(int q = commands[p][0]-1; q < commands[p][1]; q++)
            {
                temp[cnt] = array[q];
                cnt++;
                System.out.print(array[q] + " ");
            }
            Arrays.sort(temp);
            System.out.print("\t"+temp[commands[p][2]-1]);
            System.out.println();

            a[a_index] = temp[commands[p][2]-1];
            a_index++;
        }
        answer = a;
        return answer;
    }
}

File: KthNumber/src/main/kotlin/KthNumber.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = intArrayOf()
        for(p in 0 until commands.size){
            //사실 i에서 j번째까지 숫자를 자르고 k번째수를 찾는건 i+k-1 이기때문에 j는 의미가 없음
            val i = commands[p][0]-1
            val j = commands[p][1]-1
            val k = commands[p][2]-1

            var arr = intArrayOf()
            for(q in i..j){
                arr += array[q]
            }
            arr.sort()
            answer += arr[k]
        }
        return answer
    }
}